src/constants/media.ts

```diff
@@ -0,0 +1,37 @@
+/* Media types */
+
+export const AUDIO = 'audio'
+export const IMAGE = 'image'
+export const VIDEO = 'video'
+export const ALL_MEDIA = 'all'
+
+/**
+ * real media types that the API supports;
+ * - `ALL_MEDIA` is not an option here.
+ * - These types also support custom filters.
+ * - `IMAGE` should always be first here.
+ */
+export const supportedMediaTypes = [IMAGE, AUDIO]
+
+export type SupportedMediaType = typeof IMAGE | typeof AUDIO
+
+/**
+ * the types of content that users can search; `ALL_MEDIA` is also a valid
+ * option here.
+ */
+export const supportedSearchTypes = [ALL_MEDIA, IMAGE, AUDIO]
+
+export type SupportedSearchType = typeof ALL_MEDIA | SupportedMediaType
```

Comment:
Same situation here.

```suggestion
export const supportedSearchTypes = [ALL_MEDIA, IMAGE, AUDIO] as const

export type SupportedSearchType = typeof supportedSearchTypes[number]
```